monitoring/monitoring.go

```diff
@@ -0,0 +1,58 @@
+package monitoring
+
+import (
+	"google.golang.org/grpc/codes"
+)
+
+type RpcType string
+
+const (
+	Unary     RpcType = "unary"
+	Streaming RpcType = "streaming"
+)
+
+// RpcMonitor is a per-RPC datastructure
+type RpcMonitor interface {
+	// ReceivedMessage is called on every stream message received by the monitor.
+	ReceivedMessage()
+
+	// SentMessage is called on every stream message sent by the monitor.
+	SentMessage()
+
+	// Handled is called whenever the RPC handling completes (with OK or AppError).
+	Handled(code codes.Code)
+
+	// Errored is called whenever the RPC failed due to RPC-layer errors.
+	Erred(err error)
```

Comment:
other things related to what happened to the RPC. `Handled`, `Received` `Sent`. `Erred` seemed to fit.